core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/StringTerms.java

```diff
@@ -67,6 +67,10 @@ public Object getKey() {
         @Override
         public Number getKeyAsNumber() {
             // this method is needed for scripted numeric aggs
+            try {
+                return Long.parseLong(termBytes.utf8ToString());
+            } catch (NumberFormatException numberFormatException) {
```

Comment:
I think it's best to name the exception as `ignored` here; it makes it clearer that the swallowing is intentional (and some IDEs complain about unused exceptions unless they are named `ignored`).